backend/src/test/java/de/neuefische/tg/ravensperch/backend/services/KonvaServiceTest.java

```diff
@@ -0,0 +1,118 @@
+package de.neuefische.tg.ravensperch.backend.services;
+
+import de.neuefische.tg.ravensperch.backend.db.KonvaDB;
+import de.neuefische.tg.ravensperch.backend.model.CircleDto;
+import de.neuefische.tg.ravensperch.backend.model.KonvaStateDto;
+import de.neuefische.tg.ravensperch.backend.model.RectDto;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.*;
+
+class KonvaServiceTest {
+
+    private final KonvaDB testDb = mock(KonvaDB.class);
+    private final KonvaService testService = new KonvaService(testDb);
+
+    @Test
+    public void updateStateTest() {
+        // GIVEN
+        KonvaStateDto konvaStateDto = generateKonvaState();
+        when(testDb.save(konvaStateDto)).thenReturn(generateKonvaState());
+
+        // WHEN
+        Optional<KonvaStateDto> actual = testService.updateKonvaState(konvaStateDto);
+
+        // THEN
+        assertThat(actual.get(), is(generateKonvaState()));
+        verify(testDb).save(generateKonvaState());
+
+    }
+
+    @Test
+    public void getStateTest() {
+        // GIVEN
+        when(testDb.findById("konvaState")).thenReturn(Optional.of(generateKonvaState()));
+
+        // WHEN
+        Optional<KonvaStateDto> actual = testService.getKonvaState("konvaState");
+
+        // THEN
+        assertThat(actual.get(), is(generateKonvaState()));
+        verify(testDb).findById("konvaState");
+
+    }
+
```

Comment:
I would maybe expect another test for when a `konvaStateDto` cannot be found in the data base and e.g. an error is thrown or an empty Optional is returned